src/track/beatgrid.cpp

```diff
@@ -184,13 +190,49 @@ double BeatGrid::findNthBeat(double dSamples, int n) const {
         n = n + 1;
     }
 
-    double dResult = dClosestBeat + n * m_dBeatLength;
+    double dResult = floor(dClosestBeat + n * m_dBeatLength);
     if (!even(static_cast<int>(dResult))) {
         dResult--;
     }
     return dResult;
```

Comment:
yes, but I'd prefer not to change the API right now -- I would have to add static_cast<double> all over the place in the call sites.